### PR TITLE
DPD-17 add thank you page client events context and identify hits context

### PR DESCRIPTION
**Pull request by andreivaleanu**
Add externalids to events that were missing it.
thank you page 
identify

### CODE REVIEW

File: src/segmentTracker/helpers.ts [hunk 1]
```diff
@@ -17,7 +17,16 @@ const getOnetrustActiveGroups = () => {
 };
 
 export const getContext = (): Promise<LooseObject> => {
-	const userId = window.LittledataLayer?.customer?.id;
+	let userId = window.LittledataLayer?.customer?.id;
+
+	const { user } = window.analytics;
+
+	if (!userId) {
+		userId = window.Shopify.checkout.customer_id;
```
Comment (edwardupton) [inline src/segmentTracker/helpers.ts:25]: Needs conditional operators. `Shopify.checkout` won't be defined outside of checkout

File: src/segmentTracker/helpers.ts [hunk 2]
```diff
@@ -17,7 +17,16 @@ const getOnetrustActiveGroups = () => {
...
+	}
+	if (!userId && typeof user === 'function') {
+		userId = user().id();
```
Comment (edwardupton) [inline src/segmentTracker/helpers.ts:28]: Are you sure `analytics.user().id()` is not the anonymous ID?

File: src/segmentTracker/helpers.ts [hunk 3]
```diff
@@ -17,7 +17,16 @@ const getOnetrustActiveGroups = () => {
 };
 
 export const getContext = (): Promise<LooseObject> => {
-	const userId = window.LittledataLayer?.customer?.id;
+	let userId = window.LittledataLayer?.customer?.id;
```
Comment (edwardupton) [inline src/segmentTracker/helpers.ts:20]: Can you call it `shopifyCustomerId` for clarity? `userId` in the context of Segment could be another identifier: https://segment.com/docs/connections/sources/catalog/libraries/website/shopify-littledata/#user-identity

**Review by edwardupton — CHANGES_REQUESTED** [review]

File: src/segmentTracker/helpers.ts (hunk 2 above)
Comment (edwardupton) [reply src/segmentTracker/helpers.ts:28]: Either way let's drop this, as the Segment userId may not be the Shopify customer Id.

Comment (andreivaleanu) [reply src/segmentTracker/helpers.ts:28]: > Are you sure `analytics.user().id()` is not the anonymous ID?

it's not, i've checked.

Comment (andreivaleanu) [reply src/segmentTracker/helpers.ts:28]: > Either way let's drop this, as the Segment userId may not be the Shopify customer Id.

ok. will make the changes now.

**Review by edwardupton — APPROVED** [review]

**Pull request merged**